documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,42 @@
 // Module included in the following assemblies:
 //
-// assembly-oauth.adoc
+// assembly-oauth-authentication.adoc
 
 [id='con-oauth-authentication-flow-{context}']
-= OAuth 2.0 authentication mechanism
+= OAuth 2.0 authentication mechanisms
 
-The Kafka _SASL OAUTHBEARER_ mechanism is used to establish authenticated sessions with a Kafka broker.
+You can configure OAuth 2.0 authentication using one of the following authentication mechanisms:
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+* SASL_OAUTHBEARER
+* SASL_PLAIN
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+However, the authentication flow between a client, the authorization server, and the Kafka broker is different for each mechanism.
```

Comment:
I think it is really important to explain the differences and recommendations. @mstruk can provide more details, but AFAIU:
* Whenever possible, OAUTHBEARER should be used
* Only when the user has clients not supporting OAUTHBEARER, they should consider using PLAIN.

Reply:
That would be great if @mstruk can add those guidelines.

Reply:
we have a warning about using SASL plain in `KafkaClientAuthenticationPlain` schema reference